src/ui/RosterItem.ts

```diff
@@ -132,15 +161,45 @@ export default class RosterItem {
          let unreadMessages = this.contact.getTranscript().getNumberOfUnreadMessages();
          if (unreadMessages > 0) {
             this.element.addClass('jsxc-bar--has-unread-msg');
+            if (Client.getOption(SHOW_TAGS,false))
+            {
+                this.element.find('.jsxc-avatar').addClass('jsxc-bar--has-unread-msg-blink');
+            }
          } else {
             this.element.removeClass('jsxc-bar--has-unread-msg');
+            if (Client.getOption(SHOW_TAGS,false))
+            {
+                this.element.find('.jsxc-avatar').removeClass('jsxc-bar--has-unread-msg-blink');
+            }
          }
       };
 
       this.contact.getTranscript().registerHook('unreadMessageIds', updateUnreadMessage);
       updateUnreadMessage();
    }
 
+   public static convertGroupsToHtml(groups : string[]) : string
+   {
+       if (!groups)
+          return '';
+
+       let result=[];
+       let i=0
+       for (;i<groups.length;i++)
+       {
+           let indexgroup=RosterItem.allgroups.indexOf(groups[i]);
+           if (indexgroup===-1)
+           {
+               RosterItem.allgroups.push(groups[i]);
+               indexgroup=RosterItem.allgroups.length-1;
+               RosterItem.allgroupscolors.push(Color.generate(groups[i]));
+           }
+           let style='background-color: '+RosterItem.allgroupscolors[indexgroup]+';';
+           result.push('<span class="jsxc-bar-tags-item" style="'+style+'">'+groups[i]+'</span>');
+       }
+       return result.join('');
```

Comment:
For security reasons I would prefer to generate some elements instead of a string.

Reply:
changed